docs: updates for storefront in cloud (#14491)
medusajs/medusa: www/apps/cloud/app/storefront/page.mdx

import { CodeTab, CodeTabs } from "docs-ui"

export const metadata = {
  title: `Storefront Deployment on Cloud`,
}

# {metadata.title}

In this guide, learn how to deploy your storefront with your Medusa application on Cloud.

## Storefront Deployment Overview

Medusa allows you to deploy your storefront alongside your Medusa application on Cloud, enabling you to host and manage both backend and frontend in a single platform.

To deploy a storefront with your Medusa application on Cloud, you need a monorepo structure where both the Medusa application and storefront code are in the same repository. Medusa then builds and deploys both the backend and frontend.

![Diagram showcasing how a monorepo is deployed on Cloud. A project will hold a Medusa application with a server and admin, and a storefront.](https://res.cloudinary.com/dza7lstvk/image/upload/v1766044505/Cloud/storefront-overview_ebmefz.jpg)

### Storefront Deployment Per Environment

> Similar to the Medusa backend, Medusa deploys separate instances of your storefront for every long-lived and preview environment, based on your plan's limits. This allows you to test and preview changes in isolation.

Reviewer: ```suggestion
Similar to the Medusa backend, Medusa deploys a storefront for every long-lived and preview environment, based on your plan's limits. This allows you to test and preview changes in isolation.
```

1. **End-to-end deployment of your commerce application**: Host and manage all components of your commerce application, including server, admin, and storefront, in a single platform.
2. **Zero-configuration deployment**: Medusa automatically configures your server and storefront settings, including URLs and CORS, to ensure seamless connectivity between the two.
3. **Storefront previews**: Medusa creates both backend and storefront previews for GitHub pull requests, automatically configuring preview URLs and settings to connect to the corresponding Medusa preview. This allows you to preview storefront and backend changes together.
4. **Framework agnostic**: Deploy storefronts built with popular frameworks like Next.js and Tanstack Start. Refer to the [Supported Storefront Frameworks](#supported-storefront-frameworks) section for the full list of supported frameworks.
5. **Custom domains**: Set up custom domains for your storefronts to use branded URLs for your customers.
6. **SSL Certificates**: Cloud automatically provisions and renews SSL certificates for your custom domains, ensuring secure connections for your customers.

> 7. **Global CDN**: Storefronts are served via a global Content Delivery Network (CDN), ensuring fast load times for your customers regardless of their location.

Reviewer: It's a technicality, but for server-side rendered responses (eg. for NextJS) you need a server. The server also runs very close to the user on an edge node (similar to Vercel on Edge), but it's not a CDN per-se

I think you are better fitted to decide how to reword this, just wanted to bring it up
